### ISSUE TITLE
Game compatibility issues with XInput HID Device Mode

### ISSUE THREAD

**Issue by Kanuan** [ViGEm/DsHidMini#113]
## Summary
In the new `XInput HID Device Mode` introduced in DshidMini v2.0.242, the controller acts as a DInput device with a XInput compatibility layer added on top of it. It works more or less as the following:

- Games that **don't** support generic controllers will recognize the DS3 as just an Xbox One controller
- Most games that do support both generic and XInput controllers should detect that both the DInput and Xbox controller are supposed to be one, therefore ignoring the DInput part of it (e.g.: Blasphemous)

The problem is that some games will still detect the DS3 controller in `XInput HID Device Mode` as 2 separate devices, won't detect the controller at all or will detect but will have a broken layout.

## Reproducing the issue
- Set the controller in `XInput HID Device Mode`
- Use the controller in one of the affected games

## Known affected games
See [the discussion topic here](https://github.com/ViGEm/DsHidMini/discussions/114)

## Ways around the issue / Notes on what has been tested
- For games that detect the controller as 2 separater devices probably the only way around this is by .dll/api hooking that removes DInput support (not tested)
- It's impossible to hide the DInput part of the controller with HidHide since the controller doesn't appear at all on HidHide's device list
- HidGuardian was not tested
- When the game detects the controller as 2 separate devices Steam's `Xbox Configuration Support`/`Generic Configuration Support` does not prevent the issue from happening

**Comment by nefarius** [comment]
I don't think this deserves an issue. This approach is experimental from the very beginning, so whatever is not working is to be discovered via discussions and if concrete details bubble up on what can exactly be improved upon the driver, that would then warrant an issue, but not really a general "some games something something", I was aware we're not gonna have a clean solution due to the sheer amount of shit game engines out there. My 2 cents for now 😜

**Comment by nefarius** [comment]
Another thought that came up: it's nice and helpful that we start collecting information about which games work and which are troubled, but **as a developer what am I supposed to do with this information**? I'll never have all the mentioned games myself to test neither the time to invest. So, what would be the end goal? A database of "if this game better use DS4Windows route"? The new approach may not be as universally perfect as simply emulating the good old x360 pad (which is **impossible** natively with DSHMs architecture) but it opens new ways to explore how XInput is processed in games and what to do to improve it.

So TL;DR: what to do with this information and what can be given to the end-user to easy diagnostics?

**Comment by Kanuan** [comment]
> I don't think this deserves an issue. This approach is experimental from the very beginning, so whatever is not working is to be discovered via discussions and if concrete details bubble up on what can exactly be improved upon the driver

I moved all the relevant info to the [discussion topic](https://github.com/ViGEm/DsHidMini/discussions/114)

I answered your other thoughts there too.

**Comment by BlueStarBRS** [comment]
I swear, it's pure horror that HidHide doesn't show the DsHidMini input devices. That's literally the biggest reason not to use DsHidMini to finally replace the SCP Toolkit. There are other scenarios, too, where this just sucks  like in games that detect both, or when you really need to hide your input sent from DsHidMini (like in my case with a modular controller emulation translation layer)
(discord joined... waiting for verification to find a modern solution▸#newbie-8487)
